fili-core/src/main/java/com/yahoo/bard/webservice/data/dimension/AvroDimensionRowParser.java

```diff
@@ -0,0 +1,192 @@
+            throw new IllegalArgumentException(msg);
+        }
+
+        // Populating the set of avro field names
+        Set<String> avroFields = StreamSupport.stream(jsonAvroSchema.spliterator(), false)
+                .map(jsonNode -> jsonNode.get("name"))
+                .map(JsonNode::asText)
+                .collect(Collectors.toSet());
+
+        // True only if all of the mapped dimension fields are present in the Avro schema
+        return dimension.getDimensionFields().stream()
+                .map(dimensionField -> dimensionFieldMapper.convert(dimension, dimensionField))
+                .allMatch(avroFields::contains);
+    }
+
+    /**
+     * Parses the avro file and populates the dimension rows after validating the schema.
+     *
+     * @param dimension The dimension object used to configure the dimension
+     * @param avroFilePath The path of the AVRO data file (.avro)
+     * @param avroSchemaPath The path of the AVRO schema file (.avsc)
+     *
+     * @throws IllegalArgumentException thrown if JSON object `fields` is not present
+     * @throws IOException thrown if there is error parsing the avro files
+     */
+    public void parseAvroFileDimensionRows(Dimension dimension, String avroFilePath, String avroSchemaPath)
+        throws IllegalArgumentException, IOException {
+
+        if (isSchemaValid(dimension, avroSchemaPath)) {
+
+            // Creates an AVRO schema object based on the parsed AVRO schema file (.avsc)
+            Schema schema = new Schema.Parser().parse(new File(avroSchemaPath));
+
+            // Creates an AVRO DatumReader object based on the AVRO schema object
+            DatumReader<GenericRecord> datumReader = new GenericDatumReader<>(schema);
+
+            // Creates an AVRO DataFileReader object that reads the AVRO data file one record at a time
+            DataFileReader<GenericRecord> dataFileReader = new DataFileReader<>(new File(avroFilePath), datumReader);
+
+            // Generates a set of dimension Rows after retrieving the appropriate fields
+            Set<DimensionRow> dimensionRows = StreamSupport.stream(dataFileReader.spliterator(), false)
+                    .map(genericRecord -> dimension.getDimensionFields().stream().collect(
+                             Collectors.toMap(
+                                 DimensionField::getName,
+                                 dimensionField -> genericRecord.get(
+                                     dimensionFieldMapper.convert(dimension, dimensionField)
+                                 ).toString()
+                             )
+                         )
+                    )
+                    .map(dimension::parseDimensionRow)
+                    .collect(Collectors.toSet());
+
+                // Adds all dimension rows to the dimension cache
+                dimension.addAllDimensionRows(dimensionRows);
+        } else {
+            String msg = "The AVRO schema file is corrupted / empty {}";
```

Comment:
I don't think this message is correct. We hit this block if the schema isn't _valid_ (ie. doesn't have the needed fields in it), not if the file is corrupted or empty.